src/session.rs

```diff
@@ -398,11 +391,28 @@ impl Session {
                 media_need_check_source = true;
             }
         }
-        if let Some(repairs) = ssrc_repairs {
-            if source.set_repairs(repairs) {
-                media_need_check_source = true;
+
+        // Gymnastics to appease the borrow checker.
+        let source = if is_rtx {
+            // For RTX, we will look for the associated repair ssrc using the RID, if we have one.
+            // Prefer the RID repair header, if supplied, but fallback to the last RID values for
+            // the source.
+            let rid_repair = header.ext_vals.rid_repair.or_else(|| source.rid());
+            let repairs = media.ssrc_rx_for_rid(rid_repair, ssrc);
```

Comment:
Nit: this was previously called ssrc_repairs.  I'd prefer "rtx_ssrc" myself, but if we stay consistent with the existing naming "ssrc_repairs" is the way to go, I guess.

Reply:
Forget the above.  I think we should name things something like this:

```
let rtx_ssrc = ssrc;
let rtx_source = &source;
let rid = header.ext_vals.rid_repair.or_else(|| rtx_source.rid());
let primary_ssrc = find_primary_ssrc(rid, rtx_ssrc);
let primary_source = media.get_or_create_source_rx(primary_ssrc);
if let Some(primary_ssrc) = primary_ssrcy {
  if rtx_source.set_repairs(primary_ssrc) {
    media_need_check_source = true;
  }
} else if primary_source.repairs().is_none() {
 ...
}
source
```

That would be a lot more clear.

Reply:
I made updates before I saw this.. let me see how it fits

Reply:
ok... I think this isn't needed in the newest code.

"rid" is determined in the same place for rtx and non-rtx... we just look for a different header. We set the "rid" on the ReceiverSource if it's ever set.

This code therefore simplifies to just use `find_primary_ssrc_for_rid(rid, ssrc)` . I did rename it to call it `primary_ssrc` and the "else" clause doesn't need the conditional... the whole block is rtx, and find_primary_ssrc either finds a repain ssrc, or it doesn't. if it doesn't, we can always bail out.

The `source` is always the source the packet was for. we don't use the `primary_source`, and never did, this code calls `get_or_create_source_rx()` in multiple places to get around the borrow checker, since it returns a mutable pointer to the source object. you can't later access "media" immutably, without reacquiring the pointer to the source. It's the "gymnastics" the comments refer to. That seems like a separate thing to potentially refactor/work around.